src/network-config.js

```diff
@@ -114,49 +109,33 @@ export const networkConfigs = {
       portisDappId ? { id: 'portis', conf: portisDappId } : null,
     ].filter(p => p),
   },
-  unknown: {
+
+  // Local development testnet
+  local: {
     addresses: {
-      ensRegistry: localEnsRegistryAddress,
+      ensRegistry: providedEnsRegistryAddress,
     },
-    nodes: {
-      defaultEth: 'ws://localhost:8545',
+    endpoints: {
+      read: providedEthEndpoint || 'ws://localhost:8545',
     },
-    settings: {
-      name: `Unknown network`,
-      shortName: 'Unknown',
-      type: 'unknown',
+    environment: {
+      // Local development environments by convention use
+      // a chainId of value 1337, but for the sake of configuration
+      // we expose a way to change this value.
+      chainId: Number(getLocalChainId()),
+      name: 'local development',
+      shortName: 'local',
+      type: 'private',
       live: false,
     },
     providers: [{ id: 'provided' }, { id: 'frame' }],
   },
 }
 
-export function getNetworkConfig(type) {
-  return (
-    networkConfigs[type] || {
-      ...networkConfigs.unknown,
-      settings: {
-        ...networkConfigs.unknown.settings,
-        name: `Unsupported network (${type})`,
-      },
-    }
-  )
-}
-
-export function getNetworkByChainId(chainId = -1) {
-  chainId = Number(chainId)
-  return (
-    Object.values(networkConfigs).find(
-      network => network.settings.chainId === chainId
-    ) || networkConfigs.unknown
-  )
-}
-
-export function sanitizeNetworkType(networkType) {
-  if (networkType === 'private') {
-    return 'localhost'
-  } else if (networkType === 'main') {
-    return 'mainnet'
+export function getNetworkConfig(id) {
+  const networkConfig = networkConfigurations[id]
+  if (!networkConfig) {
+    throw new Error(`Unsupported network '${id}'`)
```

Comment:
Note that I've opted to entirely remove the `unsupported network` configuration.

It's impossible to statically know what chain id to expect in that case, and it's likely better the app just blows up entirely than pretend things are OK.